pkgs/development/coq-modules/mathcomp/default.nix

```diff
@@ -55,27 +55,47 @@ let
       derivation = mkCoqDerivation ({
         inherit version pname defaultVersion release releaseRev repo owner;
 
-        nativeBuildInputs = optional withDoc graphviz;
+        nativeBuildInputs = optionals withDoc [ graphviz lua ];
         mlPlugin = versions.isLe "8.6" coq.coq-version;
-        extraBuildInputs = [ ncurses which ] ++ optional withDoc lua;
+        extraBuildInputs = [ ncurses which ];
         propagatedBuildInputs = mathcomp-deps;
 
         buildFlags = optional withDoc "doc";
 
         preBuild = ''
           patchShebangs etc/utils/ssrcoqdep || true
+          patchShebangs etc/buildlibgraph || true
```

Comment:
Why should that fail?

Reply:
I'm not sure old versions of mathcomp contain this script, in which case `make doc` would not rely on it. I did not check though. Moreover if in the future we start using another tool to build the graph (or remove this one), the build would still go through, hence I prefer to leave the possibility for this patch to fail, while proceeding to the next steps in the build.

Reply:
> the build would still go through

That should be avoided. It could silently break the package and cost a lot more debugging afterwards.

Reply:
OK, right, I pushed a more robust script